scripts/reporting/batch_script_utils.py

```diff
@@ -0,0 +1,81 @@
+from subprocess import run
+
+
+def rehydrate_issue_body(body: str) -> dict:
+    return {
+        x.split(":", 1)[0].strip(): x.split(":", 1)[1].strip()
+        for x in body.split("\n-")[1:]
+    }
+
+
+def extract_unique_subject_ids(text: str) -> list:
+    subject_id_regex = "\w-\d{5}-\w-\d"
+    subject_ids = list(set(re.findall(subject_id_regex, text)))
+    return subject_ids
+
+
+def prompt_y_n(prompt: str) -> bool:
+    while True:
+        confirm = input(prompt + "\n")
+        if confirm in ["n", "y"]:
+            return confirm == "y"
+        print("Invalid input")
+
+
+def run_command(command: list, verbose: bool):
+    if verbose:
+        print(" ".join(command))
+    completed_process = run(command, capture_output=True)
+    if verbose:
+        print(f"stdout:\n{completed_process.stdout}")
+        print(f"\nstderr:\n{completed_process.stderr}")
+    return completed_process
+
+
+def prompt_close_or_comment(issue, close_comment: str):
+    if prompt_y_n("Close issue? (y/n)"):
+        issue.create_comment(close_comment)
+        issue.edit(state="closed")
+    elif prompt_y_n("Comment on issue? (y/n)"):
+        comment = input("Enter comment:\n")
+        issue.create_comment(comment)
+
+
+def get_base_command(label):
```

Comment:
I'm curious - would it make sense to allow for templating of these commands, so that instead of just knowing that you have to append an EID at the end of `check_new_sessions` and a study ID at the end of `import_mr_sessions`, you have `"{study_id}"` just waiting for a `.format` call? Just a thought - actual implementation of formatting within a list might be needlessly complicated.

Reply:
Yes, this probably makes sense especially if we want to later extend this to scripts which require multiple types of identifying information (such as update_visit_data, which also needs a --form if you don't want it to take forever). Not sure mechanically how to do this. I guess get_base_command could return a string, which is then .formatted and then split into a list?

Reply:
I think that would work, but let's table this for now - it can be a future enhancement